src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,177 @@
+    using Microsoft.ServiceFabric.Services.Remoting;
+    using Microsoft.ServiceFabric.Services.Remoting.Client;
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Fabric;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Threading.Tasks;
+    using System.Xml;
+
+    /// <summary>
+    /// Service remoting client that wraps another service remoting client and adds correlation id and context propagation support. This allows
+    /// traces the client and the service to log traces with the same relevant correlation id and context.
+    /// </summary>
+    public class CorrelatingServiceRemotingClient : IServiceRemotingClient, IWrappingClient
+    {
+        private Uri serviceUri;
+        private Lazy<DataContractSerializer> baggageSerializer;
+        private TelemetryClient telemetryClient;
+        private ServiceContext serviceContext;
+        private ITelemetryInitializer fabricTelemetryInitializer;
+
+        /// <summary>
+        /// Initializes the <see cref="CorrelatingServiceRemotingClient"/> object. It wraps the given inner client object for all the core
+        /// remote call operation.
+        /// </summary>
+        /// <param name="innerClient">The client object which this client wraps.</param>
+        /// <param name="serviceUri">The target Uri of the service which this client will call.</param>
+        public CorrelatingServiceRemotingClient(IServiceRemotingClient innerClient, Uri serviceUri)
+            : this(innerClient, serviceUri, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes the <see cref="CorrelatingServiceRemotingClient"/> object. It wraps the given inner client object for all the core
+        /// remote call operation. It allows passing in a ServiceContext object associated with the caller so telemetry can be logged with
+        /// service fabric properties.
+        /// </summary>
+        /// <param name="innerClient">The client object which this client wraps.</param>
+        /// <param name="serviceUri">The target Uri of the service which this client will call.</param>
+        /// <param name="clientServiceContext">The service context of the caller creating this client object. Pass in null if there isn't one.</param>
+        public CorrelatingServiceRemotingClient(IServiceRemotingClient innerClient, Uri serviceUri, ServiceContext clientServiceContext)
+        {
+            if (innerClient == null)
+            {
+                throw new ArgumentNullException(nameof(innerClient));
+            }
+            if (serviceUri == null)
+            {
+                throw new ArgumentNullException(nameof(serviceUri));
+            }
+
+            this.serviceContext = clientServiceContext;
+            this.fabricTelemetryInitializer = FabricTelemetryInitializerExtension.CreateFabricTelemetryInitializer(this.serviceContext);
```

Comment:
AppInsights model suggests that you add TelemetryInitializer through config/dependency injection or manually to the TelemetryConfiguration.Active.TelemetryInitializers array.
This gives users flexibility to alter behavior and simplifies code.

I don;t know what is the guidance around this telemetry initializer for SF customers, but I expect all of them already add this initializer, so having it here is a duplicate effort.

On the other hand, if customers did NOT add the initializer, even though you have it here, they would not have SF context for all other telemetry (e.g. traces. exceptions, http dependencies and requests).

So I believe using it here is not correct.